core/src/main/scala/akka/persistence/r2dbc/journal/JournalDao.scala

```diff
@@ -91,15 +105,27 @@ private[r2dbc] class JournalDao(journalSettings: R2dbcSettings, connectionFactor
     "(slice, entity_type, persistence_id, sequence_number, db_timestamp, writer, adapter_manifest, event_ser_id, event_ser_manifest, event_payload, deleted) " +
     "VALUES ($1, $2, $3, $4, transaction_timestamp(), $5, $6, $7, $8, $9, $10)"
 
-  def writeEvents(events: Seq[SerializedJournalRow]): Future[Unit] = {
+  def writeEvents(events: Seq[SerializedJournalRow], samePersistenceId: Boolean): Future[Unit] = {
     require(events.nonEmpty)
-    val persistenceId = events.head.persistenceId
 
-    val entityType = SliceUtils.extractEntityTypeFromPersistenceId(persistenceId)
-    val slice = SliceUtils.sliceForPersistenceId(persistenceId, journalSettings.maxNumberOfSlices)
-    val previousSeqNr = events.head.sequenceNr - 1
+    // normally it is the same persistenceId for all events, but when used from MigrationTool it can be
```

Comment:
thinko; it's same persistenceId also from MigrationTool, so I'll revert some of this